activerecord/lib/active_record/associations/singular_association.rb

```diff
@@ -43,11 +44,12 @@ def scope_for_create
           super.except!(*Array(klass.primary_key))
         end
 
-        def find_target
+        def find_target(async: false)
           if disable_joins
+            raise NotImplementedError if async
             scope.first
           else
```

Comment:
Disclaimer: new to Ruby, but it seems strange to me that `async` is not used in the `else` portion of the branch. Did we miss something (or what did I miss?)?

Reply:
In Ruby, `super` without arguments nor parantheses implicitly forward all arguments.

so:

```ruby
super.then(&:first)
```

is a shortcut for:

```ruby
super(async: async).then(&:first)
```